js/ui/date_box/ui.date_box.mask.parts.js

```diff
@@ -63,9 +63,17 @@ const getPatternGetter = (patternChar) => {
 };
 
 const renderDateParts = (text, format) => {
+    if(!format) {
+        return false;
+    }
+
     const regExpInfo = getRegExpInfo(format, dateLocalization),
         result = regExpInfo.regexp.exec(text);
 
+    if(!result) {
+        return false;
+    }
+
```

Comment:
Why use a two return? This looks bad.

Reply:
`getRegExpInfo` is a big method. It should not be executed when the format is not defined